source/functions.txt

```diff
@@ -12,193 +12,346 @@ Functions
    :depth: 2
    :class: singlecol
 
-Introduction
-------------
+.. toctree::
+   :titlesonly:
+   :hidden:
+   
+   Query MongoDB Atlas </functions/mongodb>
+   Context </functions/context>
+   Global Modules </functions/globals>
+   External Dependencies </functions/dependencies>
+   JavaScript Support </functions/javascript-support>
+
+Overview
+--------
 
-Realm Functions allow you to define and execute server-side logic for
-your application. You can call functions from your client applications
-as well as from other functions and in JSON expressions throughout
-Realm.
+A **{+service-short+} Function** is a piece of server-side JavaScript
+code that you write to define your app's behavior. You can call your
+app's functions directly from a client app or define services that
+integrate and call functions automatically.
+
+Functions can call other functions and include a built-in client for
+working with data in {+atlas+} clusters. They also include helpful
+global utilities, support common Node.js built-in modules, and can
+import and use external packages from the npm registry.
+
+Functions are Serverless
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a function is called, your app routes the request to a managed app
+server that evaluates your code and returns the result. This model makes
+functions *serverless*, which means that you don't have to deploy and
+manage a server to run the code. Instead, you write the function source
+code and your app handles the execution environment.
+
+Functions have Context
+~~~~~~~~~~~~~~~~~~~~~~
+
+A function runs in a :ref:`context <function-context>` that reflects its
+execution environment. The context includes the user that called the
+function, how they called it, and the state of your app when they called
+it. You can use context to run user-specific code and work with other
+parts of your app.
```

Comment:
i'd change this section from inline linking to `context` to including a seealso or separate sentence at the bottom of the section linking to the functions context page